cpp/add_synonyms.cc

```diff
@@ -61,10 +62,30 @@ std::string GetSubfieldCodes(const std::string &tag_and_subfields_spec) {
 }
 
 
+bool FilterPasses(const MarcRecord &record, const std::map<std::string, std::pair<std::string, std::string>> &filter_specs, std::string field_spec) {
+      auto filter_spec = filter_specs.find(field_spec);
+      if (filter_spec == filter_specs.cend())
+          return true;
+
+      auto rule = filter_spec->second;
+      // We have field_spec in key and rule to match in value
+      std::string subfield_value;
+      std::string subfield_code = GetSubfieldCodes(rule.first);
```

Comment:
Please use parentheses, also the variable is misnamed, it should be named "subfield_codes".